finality-aleph/src/network/address_cache.rs

```diff
@@ -0,0 +1,184 @@
+use std::{fmt::Debug, marker::PhantomData, num::NonZeroUsize, sync::Arc};
+
+use lru::LruCache;
+use parking_lot::Mutex;
+use primitives::{AccountId, AlephSessionApi, AuthorityId, BlockHash, BlockNumber};
+use sc_client_api::Backend;
+use sp_runtime::traits::{Block, Header};
+
+use crate::{
+    abft::NodeIndex,
+    session::{SessionBoundaryInfo, SessionId},
+    session_map::AuthorityProvider,
+    ClientForAleph,
+};
+
+pub trait KeyOwnerInfoProvider {
+    fn aleph_key_owner(&self, block_number: BlockNumber, key: AuthorityId) -> Option<AccountId>;
+}
+
+/// Network details for a given validator in a given session.
+#[derive(Debug, Clone)]
+pub struct ValidatorAddressingInfo {
+    /// Session to which given information applies.
+    pub session: SessionId,
+    /// Network level address of the validator, i.e. IP address (for validator network)
+    pub network_level_address: Option<String>,
+    /// PeerId of the validator used in validator (clique) network
+    pub validator_network_peer_id: String,
```

Comment:
> nit: change String to PeerId

Validator network `PeerId` is a trait, implemented for tcp network addressing info, so we'd have to add a parameter to that struct (and others in that file), and even worse, since this struct will be passed to rpc, we'd have to specialize network type in runtime instead of `finality-aleph`.

Reply:
ok, fair